src/engine/source/cmstore/interface/cmstore/dataintegration.hpp

```diff
@@ -119,15 +135,83 @@ class Integration
         {
             throw std::runtime_error("Integration category cannot be empty");
         }
+        if (!cm::store::categories::exists(m_category))
+        {
+            throw std::runtime_error("Integration category is not valid: " + m_category);
+        }
+
+        if (m_defaultParent.has_value())
+        {
+            if (!base::utils::generators::isValidUUIDv4(*m_defaultParent))
+            {
+                throw std::runtime_error("Integration default parent is not a valid UUIDv4: " + *m_defaultParent);
+            }
+        }
+
+        for (const auto& uuid : m_decodersByUUID)
+        {
+            if (!base::utils::generators::isValidUUIDv4(uuid))
+            {
+                throw std::runtime_error("Decoder UUID is not a valid UUIDv4: " + uuid);
+            }
+        }
+
+        for (const auto& uuid : m_kvdbsByUUID)
+        {
+            if (!base::utils::generators::isValidUUIDv4(uuid))
+            {
+                throw std::runtime_error("KVDB UUID is not a valid UUIDv4: " + uuid);
+            }
+        }
+
+        for (const auto& uuid : m_outputsByUUID)
+        {
+            if (!base::utils::generators::isValidUUIDv4(uuid))
+            {
+                throw std::runtime_error("Output UUID is not a valid UUIDv4: " + uuid);
+            }
+        }
+
+        if (auto dup = cm::store::detail::findDuplicateUUID(m_decodersByUUID); dup.has_value())
+        {
+            throw std::runtime_error("Duplicate decoder UUID: " + *dup);
+        }
+
+        if (auto dup = cm::store::detail::findDuplicateUUID(m_kvdbsByUUID); dup.has_value())
+        {
+            throw std::runtime_error("Duplicate KVDB UUID: " + *dup);
+        }
+
+        if (auto dup = cm::store::detail::findDuplicateUUID(m_outputsByUUID); dup.has_value())
+        {
+            throw std::runtime_error("Duplicate output UUID: " + *dup);
+        }
+
         updateHash();
     }
 
-    static Integration fromJson(const json::Json& integrationJson)
+    static Integration fromJson(const json::Json& integrationJson, bool requireUUID)
```

Comment:
I see that this fromJson handles the requireUUID parameter differently compared to kvdb, why is that?